File: LeetCode-Algorithm/0001. Two Sum/Solution.kt
package kt

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        for (i in nums.indices) {
            for (j in nums.indices) {
                if (i == j)
                    continue
                if (nums[i] + nums[j] == target)
                    return intArrayOf(i, j)
            }
        }

        return intArrayOf(0, 0)
    }
}

File: LeetCode-Algorithm/0001. Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target)
    {
        unordered_map<int, int> hash; //Number, Index

        for (int i = 0; i < numbers.size(); i++)
        {
            int numberToFind = target - numbers[i];

            if (hash.find(numberToFind) != hash.end())
                return  {hash[numberToFind], i};

            hash[numbers[i]] = i;
        }
        return {};
    }
};
